src/xdocs/checks/blocks/needbraces.xml

```diff
@@ -88,24 +88,50 @@
         </source>
         <p id="Example1-code">Example:</p>
         <source>
-if (obj.isValid()) return true; // violation, single-line statements not allowed without braces
-if (true) {                     // OK
+class Example1 {
+  Helper obj = new Helper();
+  public boolean Example1() {
+    int num = 12;
+    if (obj.isValid(num)) return true; // violation
```

Comment:
Example become more complicated.
Can we reuse https://docs.oracle.com/javase/8/docs/api/java/lang/String.html as object and call some methods in this code just vto make it compile-able, and it will continue to be nonsense in meaning but, it is fine all we care here are braces. It is ok that there will be more diff, but users will have better experience.

Please also restore `// violation, single-line statements not allowed without braces` to explain user why it is violation to help user easily catch point of attention.

Reply:
If you are talking about any random method to call then we can do . Let me try implementing this as well !

Reply:
> Please also restore `// violation, single-line statements not allowed without braces` to explain user why it is violation to help user easily catch point of attention.

Sure !

Reply:
> // violation, single-line statements not allowed without braces

Done !

Reply:
> Can we reuse https://docs.oracle.com/javase/8/docs/api/java/lang/String.html as object

done !